### PR TITLE
Callback on network click

### PR DESCRIPTION
**Pull request by punit-shah**
Allows developer to set a callback to be called when the user clicks on a sharetool.

Thanks for the help @fewstera 🙏

### CODE REVIEW

File: tests/controllerSpec.js [hunk 1]
```diff
@@ -77,6 +77,15 @@ define(['ShareTools'], function (ShareToolsController) {
...
 
+        it('should let me set a callback to be called when a network is clicked', function () {
+            var shareClickedSpy = jasmine.createSpy('shareClickedSpy');
+            controller.onShareButtonClick(shareClickedSpy);
+
+            controller.satisfyShareButtonCallback('facebook');
+            expect(shareClickedSpy).toHaveBeenCalled();
+            expect(shareClickedSpy).toHaveBeenCalledWith('facebook');
+        });
+
```
Comment (fewstera) [inline tests/controllerSpec.js:91]: @punit-shah One thing we missed - testing multiple callbacks.

Would be nice to create two spys, add them as callbacks, ensure they havent been called, then statifying and then finally check they've both been called.

Comment (punit-shah) [inline tests/controllerSpec.js:91]: What does it mean to statify?

**Comment by punit-shah** [comment]
@fewstera 🚨 test for multiple callbacks written

**Comment by fewstera** [comment]
Nice work @punit-shah. Happy to merge 👍

**Pull request merged**